migrate/migrations/20190613111645-add-rotation-favorite.sql

```diff
@@ -0,0 +1,17 @@
+-- +migrate Up
+ALTER TABLE user_favorites ADD COLUMN tgt_rotation_id UUID REFERENCES rotations (id);
+
+ALTER TABLE user_favorites ALTER COLUMN tgt_service_id DROP NOT NULL;
+
+ALTER TABLE user_favorites ADD CONSTRAINT user_fav_rotations UNIQUE (user_id, tgt_rotation_id);
```

Comment:
I think we can put these all into a single statement. That way we only run a single round-trip to Postgres while holding the table lock instead of 3.

As for the name, these have to be globally unique. I'd suggest just going with the Postgres default: `user_favorites_user_id_tgt_rotation_id_key` to match what we have for services.

```sql
-- Table Definition ----------------------------------------------

CREATE TABLE user_favorites (
    user_id uuid NOT NULL REFERENCES users(id) ON DELETE CASCADE,
    tgt_service_id uuid NOT NULL REFERENCES services(id) ON DELETE CASCADE,
    id BIGSERIAL UNIQUE,
    CONSTRAINT user_favorites_user_id_tgt_service_id_key UNIQUE (user_id, tgt_service_id)
);

-- Indices -------------------------------------------------------

CREATE UNIQUE INDEX user_favorites_user_id_tgt_service_id_key ON user_favorites(user_id uuid_ops,tgt_service_id uuid_ops);
CREATE UNIQUE INDEX user_favorites_uniq_id ON user_favorites(id int8_ops);
```